.pre-commit-config.yaml

```diff
@@ -64,6 +64,15 @@ repos:
     hooks:
       - id: yesqa
 
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.2
+    hooks:
+      - id: codespell
+        args: ["--write-changes"]
+        exclude: "extern|cextern"
```

Comment:
This line is partially redundant with the `skip` field in the `codespell` configuration in `pyproject.toml`, but appears necessary.  The `pyproject.toml` config is already set to skip `*extern`. However, running `pre-commit run --all-files` did not exclude `extern` and `cextern` from getting processed by codespell.  Weird.

We still want to keep the `skip` field in `pyproject.toml` as-is, since that gives contributors the option to run codespell outside of pre-commit.

Reply:
This is necessary due to the peculiarities of pre-commit. Basically, the way pre-commit works is that it passes each file tracked by git individually to the tool in question rather letting the tool itself iterate over all the files. This is so that pre-commit can pull tricks like running only on files staged for commit rather than having to run on the whole repo.

Often tools are written in a way so that if you explicitly pass a file which it is configured to skip, it will still run on that file. I suspect codespell works that way. Some tools like black have a skip/exclude (force-exclude) that you can set which will never be overridden in this fashion. I wonder if codespell has a similar mechanism?

Reply:
Yes. I wish pre-commit had an option to grab the ``exclude`` settings from the tools' configs, like

``exclude: pyproject.toml::codespells::exclude`` so there were only 1 source of configuration. But alas.